### PR TITLE
GEA2139 Tamper proof updates

### PR DESCRIPTION
**Pull request by pangea-andrest**

### CODE REVIEW

File: service/audit/api.go [hunk 1]
```diff
@@ -186,38 +183,69 @@ func SearchAll(ctx context.Context, client Client, input *SearchInput) (*Root, S
...
+func (a *Audit) processLogResponse(ctx context.Context, log *LogOutput) error {
+	if log == nil {
+		return nil
 	}
 
-	rp := NewArweaveRootsProvider(root.TreeName)
+	nurh := log.UnpublishedRootHash
+
+	if a.VerifyProofs {
+		if VerifyHash(*log.EventEnvelope, log.Hash) == Failed {
```
Comment (dbenders) [inline service/audit/api.go:197]: warning: segfault if EventEnvelope is nil (can happen?)

File: service/audit/api.go [hunk 2]
```diff
@@ -447,70 +496,95 @@ type SearchEvent struct {
...
-	if root == nil || ee.MembershipProof == "" {
-		return true
+func (ee *SearchEvent) VerifyConsistencyProof(publishedRoots map[int]Root) {
+	if ee.Published == nil || *ee.Published != true || ee.LeafIndex == nil {
+		ee.ConsistencyVerification = NotVerified
+		return
 	}
-
-	b, err := VerifyMembershipProof(*root, *ee, false)
+	idx := *ee.LeafIndex
```
Comment (dbenders) [inline service/audit/api.go:531]: ee.LeafIndex=nil?

**Review by dbenders — DISMISSED** [review]
be careful with the dereferencing of pointers

File: service/audit/api.go (hunk 1 above)
Comment (pangea-andrest) [reply service/audit/api.go:197]: Fix this one!

File: service/audit/api.go (hunk 2 above)
Comment (pangea-andrest) [reply service/audit/api.go:531]: It's checked some lines above and return if nil

**Review by dbenders — APPROVED** [review]

**Pull request merged**